lib/spark_api/authentication/oauth2_impl/grant_type_base.rb

```diff
@@ -45,7 +45,7 @@ def create_session(token_params)
           response.expires_in = provider.session_timeout if response.expires_in.nil?
           SparkApi.logger.debug { "[oauth2] New session created #{response}" }
           response
-        rescue Faraday::Error::ConnectionFailed => e
+        rescue Faraday::ConnectionFailed => e
```

Comment:
do you know offhand which version of faraday moved these?

Reply:
Faraday v1.0.0. requires this change. If we were to lock in at 0.17.3, we wouldn't need to do this.

Reply:
I don't quite follow; are you saying requiring 0.17.3 or higher (as the gemspec does) works with this, or that we need 1.0.0+?

Reply:
As it sits this works for both 0.17.3 _and_ 1.0.0+.  The old scope ( Faraday::Error::ConnectionFailed ) will break on 1.0.0.

Reply:
Interesting, so the old class location must have been deprecated, would be my guess.